UI/src/controller/laneModellingCollection.js

```diff
@@ -0,0 +1,140 @@
+
+      var collectionWithoutGroup = collectionPartitionedBySelection[false] || [];
+      var groupWithoutSelection = _.reject(groupContainingSelection, isSelected);
+
+      return collectionWithoutGroup.concat(_.isEmpty(groupWithoutSelection) ? [] : [groupWithoutSelection]).concat([selection.get()]);
+    };
+
+    var isEqual = function(a, b) {
+      function equalUnknown() {
+        return (_.has(a, 'generatedId') && _.has(b, 'generatedId') && (a.generatedId === b.generatedId));
+      }
+
+      function equalExisting() {
+        return (!_.isUndefined(a.id) && !_.isUndefined(b.id) && (a.id === b.id));
+      }
+
+      return equalUnknown() || equalExisting();
+    };
+
+    this.getGroup = function(segment) {
+      return _.find(self.linearAssets, function(linearAssetGroup) {
+        return _.some(linearAssetGroup, function(s) { return s.linkId == segment.linkId && s.sideCode == segment.sideCode; });
+      });
+    };
+
+    var replaceOneSegment = function(collection, segment, newSegment) {
+      var collectionPartitionedBySegment = _.groupBy(collection, function(linearAssetGroup) {
+        return _.some(linearAssetGroup, function(s) {
+          return isEqual(s, segment);
+        });
+      });
+      var groupContainingSegment = _.flatten(collectionPartitionedBySegment[true] || []);
+
+      var collectionWithoutGroup = collectionPartitionedBySegment[false] || [];
+      var groupWithoutSegment = _.reject(groupContainingSegment, function(s) { return isEqual(s, segment); });
+
+      return collectionWithoutGroup.concat(_.map(groupWithoutSegment, function(s) { return [s]; })).concat([[newSegment]]);
+    };
+
+    this.replaceSegments = function(selection, lane, newSegments) {
+      self.linearAssets = replaceOneSegment(self.linearAssets, lane, newSegments);
+      return newSegments;
+    };
+
+    var calculateMeasure = function(link) {
+      var points = _.map(link.points, function(point) {
+        return [point.x, point.y];
+      });
+      return new ol.geom.LineString(points).getLength();
+    };
+
+    this.splitLinearAsset = function(lane, split, callback) {
+      var left = _.cloneDeep(lane);
+      left.points = split.firstSplitVertices;
+      left.endMeasure = split.splitMeasure;
+
+      var right = _.cloneDeep(lane);
+      right.points = split.secondSplitVertices;
+      right.startMeasure = split.splitMeasure;
+
+      if (calculateMeasure(left) < calculateMeasure(right)) {
+        splitLaneAssets.created = left;
+        splitLaneAssets.existing = right;
+      } else {
+        splitLaneAssets.created = right;
+        splitLaneAssets.existing = left;
+      }
+
+      splitLaneAssets.created.id = 0;
+      splitLaneAssets.existing.id = 0;
+
+      splitLaneAssets.created.marker = 'A';
+      splitLaneAssets.existing.marker = 'B';
+
+      dirty = true;
+      callback(splitLaneAssets);
+    };
+  };
+})(this);
```

Comment:
There is some duplicated code between LaneModellingCollection.js and LinearAssetsCollection.js that could be passed through inheritance. For example multiElementEvent is the same on both, this could be changed on LinearAssetsCollection.js to:
``` JavaScript
 this.multiElementEvent = function (eventName) {
          return multiElementEventCategory + ':' + eventName;
      };
```
And used on LinearAssetsCollection.js and LaneModellingCollection.js as 
```JavaScript
 self.multiElementEvent(eventName)

Reply:
Applied